win32ss/gdi/ntgdi/freetype.c

```diff
@@ -2326,12 +2327,123 @@ IntGdiAddFontResourceEx(
     return ret;
 }
 
+/* Borrowed from shlwapi */
+static PWSTR
+PathFindFileNameW(PCWSTR lpszPath)
+{
+    PCWSTR lastSlash = lpszPath;
+    while (*lpszPath)
+    {
+        if ((*lpszPath == L'\\' || *lpszPath == L'/' || *lpszPath == L':') &&
+            lpszPath[1] && lpszPath[1] != '\\' && lpszPath[1] != L'/')
+        {
+            lastSlash = lpszPath + 1;
+        }
+        lpszPath++;
+    }
+    return (LPWSTR)lastSlash;
+}
+
+/* Delete registry font entry */
+static VOID
+IntDeleteRegFontEntry(_In_ PCWSTR pszFileName, _In_ DWORD dwFlags)
+{
+    NTSTATUS Status;
+    HKEY hKey;
+    WCHAR szName[MAX_PATH], szValue[MAX_PATH];
+    ULONG dwIndex, NameLength, ValueSize, dwType;
+
+    Status = RegOpenKey(g_FontRegPath.Buffer, &hKey);
+    if (!NT_SUCCESS(Status))
+        return;
+
+    for (dwIndex = 0; NT_SUCCESS(Status); ++dwIndex)
+    {
+        NameLength = RTL_NUMBER_OF(szName);
+        ValueSize = sizeof(szValue);
+        Status = RegEnumValueW(hKey, dwIndex, szName, &NameLength, &dwType, szValue, &ValueSize);
+        if (!NT_SUCCESS(Status))
+            break;
+
+        if (dwType != REG_SZ || _wcsicmp(szValue, pszFileName) != 0)
+            continue;
+
+        /* Delete the found value */
+        Status = RegDeleteValueW(hKey, szName);
```

Comment:
Could there be more than one value with the name of `pszFileName` ?

Reply:
I think Nope.